terminus-ui/drawer/src/drawer-container.component.ts

```diff
@@ -0,0 +1,368 @@
+   */
+  public updateContentMargins() {
+    let left = 0;
+    let right = 0;
+    left = this.start ? this.marginCalculation(this.start, 'left', 0) : left;
+    right = this.end ? this.marginCalculation(this.end, 'right', 0) : right;
+
+    // If either `right` or `left` is zero, don't set a style to the element. This
+    // allows users to specify a custom size via CSS class in SSR scenarios where the
+    // measured widths will always be zero. Note that we reset to `null` here, rather
+    // than below, in order to ensure that the types in the `if` below are consistent.
+    // tslint:disable-next-line:no-non-null-assertion
+    left = left || null!;
+    // tslint:disable-next-line:no-non-null-assertion
+    right = right || null!;
+    if (left !== this.contentMargins.left || right !== this.contentMargins.right) {
+      this.contentMargins = {
+        left,
+        right,
+      };
+
+      // Pull back into the NgZone since in some cases we could be outside.
+      this.ngZone.run(() => this.contentMarginChanges.next(this.contentMargins));
+    }
+  }
+
+  /**
+   * For drawers in `overlay` mode, they don't affect the content.
+   * For drawers in `push` mode they should shrink the content. We do this by adding to the
+   * left margin (for left drawer) or right margin (for right the drawer).
+   * If the content margin has already been set and there are more than 1 opening windows on the same size, do not update.
+   *
+   * @params - drawer component, side of the drawer and margin on that side
+   * @return - calculated margin on the specific side.
+   */
+  private marginCalculation(drawer: TsDrawerComponent | null, side: string, margin: number): number {
+    let sideMargin;
+    if (this.content) {
+      sideMargin = side === 'left' ? this.content.leftMargin : this.content.rightMargin;
+    }
+    if (drawer && drawer.isExpanded) {
+      if (drawer.mode === 'push') {
+        const width = parseFloat(drawer.expandedSize.replace(/[^0-9.]/g, ''));
+        const measurement = drawer.expandedSize.match(/[a-zA-Z]+/g) || '';
+        const coefficient = measurement.includes('px') ? 1 : MARGIN_SIZE_CONVERSION;
+        margin += width * coefficient;
+      } else if (this.content && sideMargin && this.openDrawerCount <= 1) {
+        margin = 0;
+      } else if (this.content && sideMargin && this.openDrawerCount > 1) {
+        margin = this.contentMargins[side] || 0;
+      }
+    }
+    return margin;
+  }
+
+  /**
+   * Subscribes to drawer events in order to set a class on the main container element when the
+   * drawer is open and the backdrop is visible. This ensures any overflow on the container element
+   * is properly hidden.
+   *
+   * @params drawer - TsDrawerComponent
+   */
+  private adjustDrawerOnChange(drawer: TsDrawerComponent): void {
+    drawer.animationStarted.pipe(
+      filter(event => event.fromState !== event.toState),
+      takeUntil(this.drawers.changes),
```

Comment:
I _think_ we still need `untilComponentDestroyed` here?

Reply:
My thought was that `takeUntil` would stop when `drawer` is destroyed and no changes can be emitted. So there is no need to add that. But it doesn't hurt either...

Reply:
I'm not sure that would actually stop. If drawers were de-registered during the destroy lifecycle, `this.drawers` would still be an empty query list so the subscribe would happen again.